src/app/components/qr-scan/qr-scan.component.html

```diff
@@ -0,0 +1,58 @@
+<div uk-grid>
+  <div class="uk-width-1-1 uk-animation-slide-left-small">
+
+    <div uk-grid class="uk-margin-top">
+
+      <div class="uk-width-1-1@m" [hidden]="!hasDevices">
+
+        <div class="uk-width-1-1">
+          <div class="uk-form-horizontal">
+
+            <div class="uk-margin">
+              <label class="uk-form-label">Device</label>
+              <div class="uk-form-controls">
+                <select class="uk-select" (change)="onDeviceSelectChange($event.target.value)">
+                  <option value="" [selected]="!currentDevice">No Device Selected</option>
+                  <option *ngFor="let device of availableDevices" [value]="device.deviceId"
+                    [selected]="currentDevice && device.deviceId === currentDevice.deviceId">{{device.label}}</option>
+                </select>
+              </div>
+            </div>
+
+          </div>
+        </div>
+
+        <div class="uk-card uk-card-default uk-card-small">
+          <div class="uk-card-body uk-center">
+            <zxing-scanner [torch]="torchEnabled" [(device)]="currentDevice" (scanSuccess)="onCodeResult($event)"
+              [formats]="formatsEnabled" [tryHarder]="tryHarder" (permissionResponse)="onHasPermission($event)"
+              (camerasFound)="onCamerasFound($event)" (torchCompatible)="onTorchCompatible($event)"></zxing-scanner>
+          </div>
+        </div>
+
+      </div>
+
+    </div>
+
+  </div>
+</div>
+
+<div *ngIf="hasPermission === undefined" class="uk-alert uk-alert-info">
```

Comment:
According to [this chart](https://github.com/zxing-js/ngx-scanner/wiki/%60hasPermission%60-and-%60hasDevices%60-cheat-sheet), `hasPermission` and `hasDevices` could also be `null`, so I would suggest we check for `hasPermission == null` which will cover both the null and undefined cases.

Might also be a good idea to add this the type information to the .ts file when these variables are initialized. For example (line 27): `hasDevices: boolean|null = null;`

Reply:
At the bottom of the doc there is a table with the states:
![grafik](https://user-images.githubusercontent.com/11021604/86771816-2d1e4b00-c053-11ea-970c-14950e153fe8.png)

So `undefined` is the initial state and get's updated to `null` if there is a weird error. So we should leave the initial state as it is, but I updated the test cases for the alerts. That fixes the weird bug of having two alerts at once. Thanks for the review!